feat(workspace/app_server_group): add resource to manage server group (#5817)
huaweicloud/terraform-provider-huaweicloud: docs/resources/workspace_app_server_group.md

  Changing this creates a new resource.  
  This parameter value must be the subnet ID corresponding to the Workspace service.

* `system_disk_type` - (Required, String) Specifies the type of system disk.  
  The valid values are as follows:
  + **ESSD**: Extreme SSD type.
  + **SSD**: Ultra-high I/O type.
  + **GPSSD**: General purpose SSD type.
  + **SAS**: High I/O type.
  + **SATA**: Common I/O type.

* `system_disk_size` - (Required, Int) Specifies the size of system disk, in GB.  
  The minimum value of this parameter cannot be less than the disk size corresponding to the image.

* `image_id` - (Required, String) Specifies the image ID of the server group.

* `image_type` - (Required, String) Specifies the image type of the server group.  
  The valid values are as follows:
  + **gold**: The market image.
  + **public**: The public image.
  + **private**: The private image.
  + **shared**: The shared image.
  + **other**

* `image_product_id` - (Optional, String) Specifies the image product ID of the server group.  
  This parameter is required wnen the `image_type` parameter is set to **gold**.

> * `is_vdi` - (Optional, Bool, ForceNew) Specifies the session mode of the server group. Defaults to `false`.

Reviewer: ```suggestion
* `is_vdi` - (Optional, Bool, ForceNew) Specifies the session mode of the server group. Defaults to **false**.
```

  + **false**: Multi-session mode.
  + **true**: Single-session mode.

  If the AD server is not connected, only the single-session mode is supported.  
  Changing this creates a new resource.

* `app_type` - (Optional, String) Specifies the type of application group associated with the server group.
  Defaults to **COMMON_APP**.  
  The valid values are as follows:
  + **SESSION_DESKTOP_APP**
  + **COMMON_APP**

* `availability_zone` - (Optional, String, ForceNew) Specifies the availability zone of the server group.  
  If omitted, the AZ randomly assigned by the system is used.  
  Changing this creates a new resource.

* `description` - (Optional, String) Specifies the description of the server group.

* `tags` - (Optional, Map, ForceNew) Specifies the key/value pairs to associate with the server group.
  Supports up to 20 tags.  
  Changing this creates a new resource.
  

> * `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID of the server group.  

Reviewer: ```suggestion
* `enterprise_project_id` - (Optional, String, ForceNew) Specifies the enterprise project ID to which the server group belongs.  
```

  -> If any metric of the server exceeds the threshold, new sessions will be rejected. The sessions will
     be automatically scheduled to other available servers.

* `ou_name` - (Optional, String) Specifies the OU name corresponding to the AD server.
  This parameter is available only when the AD server is connected.

* `extra_session_type` - (Optional, String, ForceNew) Specifies the additional session type.
  This parameter is available only wnen the `is_vdi` parameter is set to **false**.  
  Changing this creates a new resource.  
  The valid values are as follows:
  + **GPU**
  + **CPU**
  
* `extra_session_size` - (Optional, Int, ForceNew) Specifies the number of additional sessions for a single server.
  Changing this creates a new resource.  
  This parameter is available only wnen the `is_vdi` parameter is set to **false**.  
  The `extra_session_size` must be used together with `extra_session_type`.  
  The upper limit of the number of additional sessions for a single server is `10` times the number of vCPUs in the server
  specification minus the default number of sessions in the package.

> * `primary_server_group_id` - (Optional, String, ForceNew) Specifies the ID of the primary server group ID.

Reviewer: ```suggestion
* `primary_server_group_id` - (Optional, String, ForceNew) Specifies the ID of the primary server group.
```
